### PR TITLE
Add release-note/changelog for v0.12.0

### PR DESCRIPTION
**Pull request by sunnylovestiramisu**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design

/kind documentation

> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

I ran release note sha diff of:
```
release-notes \
  --org=kubernetes-csi \
  --repo=csi-lib-utils \
  --start-rev=release-0.11 \
  --discover=mergebase-to-latest \
  --required-author="" \
  --markdown-links \
  --output out.md
```

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: CHANGELOG/CHANGELOG-0.12.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
+# Release notes for v0.12.0
+
+# Changelog since v0.11.0
+
+## Changes by Kind
+
+### Other (Cleanup or Flake)
+
+- Leader election is only supported for leases via NewLeaderElection. ([#109](https://github.com/kubernetes-csi/csi-lib-utils/pull/109), [@pohly](https://github.com/pohly))
```
Comment (msau42) [inline CHANGELOG/CHANGELOG-0.12.md:9]: It looks like this change is already part of v0.11.0. Not sure why it got picked up by the tool. https://github.com/kubernetes-csi/csi-lib-utils/blob/master/CHANGELOG/CHANGELOG-0.11.md

Comment (sunnylovestiramisu) [reply CHANGELOG/CHANGELOG-0.12.md]: Updated!

Comment (xing-yang) [reply CHANGELOG/CHANGELOG-0.12.md]: The start sha should be f89955e5695f1d0d2dcaa45939c0e57e9ea6cf59.  The start sha used in this run was before the 0.11 release.

Comment (sunnylovestiramisu) [reply CHANGELOG/CHANGELOG-0.12.md]: I changed to use tags, this command works better:

```
release-notes \
  --org=kubernetes-csi \
  --repo=csi-lib-utils \
  --start-rev=release-0.11 \
  --discover=mergebase-to-latest \
  --required-author="" \
  --markdown-links \
  --output out.md
```

Comment (msau42) [reply CHANGELOG/CHANGELOG-0.12.md]: I think `--start-rev` is ignored if you use `--discover`.

**Pull request merged**